### DISCUSSION TITLE
Get Weather results in Unhandled Event Loop Exception

### DISCUSSION THREAD

**Discussion by stvzldn** [mytourbook/mytourbook#1343] [General]
Uploaded an Edge 520 .fit today with success.
Then went to fetch weather from the Tour Editor with no response.
Did a Save for the other entered fields of the tour.
Then attempted the weather fetch again,  Resulted in the exception below:

> 14:24:06 [main - 1] 1  Importing tours... 
14:24:06 [Importing tours - 66] 2  FIT SDK 21.126 
14:24:06 [Importing tours - 66] 3  0.650 s ∙ 10,697 # ∙ 5/13/2024, 8:49 AM ← C:\Users\sz\Desktop\2024-05-13-08-49-27.fit 
14:24:07 [main - 1] 4  Imported in 0.909 s 
14:24:08 [main - 1] 5  Save tours 
14:24:08 [main - 1] 6  Saved 1 tours in 0.181 s 
14:24:20 [main - 1] 7  Retrieving weather data with "OpenWeatherMap" 
14:24:20 [main - 1] 8  5/13/2024, 8:49 AM -> Error while retrieving the weather data: "null" 
14:24:20 [main - 1] 9  5/13/2024, 8:49 AM -> Error while retrieving the weather data: "null" 
14:24:20 [main - 1] 10  5/13/2024, 8:49 AM . . . Weather data not found 
14:24:20 [main - 1] 11  Data retrieved in 0.1 s 
14:24:31 [main - 1] 12  Retrieving weather data with "OpenWeatherMap" 

![Image1](https://github.com/mytourbook/mytourbook/assets/106929478/5deab1e0-7a3e-469f-979d-d3166ab25b7b)
[Unhandled Loop Fault.txt](https://github.com/mytourbook/mytourbook/files/15298714/Unhandled.Loop.Fault.txt)

**Comment by FJBDev** [comment]
This is a known issue that will be fixed in 24.5

See more details [here](https://github.com/mytourbook/mytourbook/issues/1337)
